code/modules/mob/_modifiers/traits_phobias.dm

```diff
@@ -441,6 +441,86 @@
 				return
 	return fear_amount
 
+/datum/modifier/trait/phobia/trypanophobe
+	name = "trypanophobia"
+	desc = "Syringes and needles make you very distressed. You really don't want to get sick..."
+	fear_decay_rate = 10
+
+	on_created_text = "<span class='warning'>You are terrified by needles.</span>"
+	on_expired_text = "<span class='notice'>You feel better about being near needles..</span>"
+
+
+	zero_fear_up = list(
+		"<span class='warning'><font size='3'>That's a needle!</font></span>",
+		"<span class='warning'><font size='3'>There's a needle right there!</font></span>"
+		)
+	zero_fear_down = list(
+		"<span class='notice'>The needle is gone, no need to worry.</span>",
+		"<span class='notice'>No more needle.</span>"
+		)
+
+	half_fear_up = list(
+		"<span class='danger'><font size='3'>The needle could get you at any time!</font></span>",
+		"<span class='danger'><font size='3'>The needle is still there!</font></span>"
+		)
+	half_fear_down = list(
+		"<span class='warning'>There are no more needles... right?</span>",
+		"<span class='warning'>You don't see any more needles... But you can never be sure.</span>"
+		)
+
+	full_fear_up = list(
+		"<span class='danger'><font size='4'>The needles are going to pierce you!</font></span>",
+		"<span class='danger'><font size='4'>They could get you any second!</font></span>"
+		)
+	full_fear_down = list(
+		"<span class='danger'>There are more needles around, you can feel it...</span>",
+		"<span class='danger'>No more needles, please...</span>"
+		)
+
+/datum/modifier/trait/phobia/trypanophobe/should_fear()
+	if(holder.blinded)
+		return 0 //Cannot feareth what cannot beest seen
+
+	var/fear_amount = 0
+
+	for(var/atom/thing in view(5, holder)) // See haemophobia for why this is 5.
+		if(istype(thing, /obj/item/weapon/reagent_containers/syringe))
+			var/obj/item/weapon/reagent_containers/syringe/S = thing
+			if(S.reagents && S.reagents.total_volume)
+				fear_amount += 8 //They're gonna give you some stabbies
+			else
+				fear_amount += 4
+
+		if(istype(thing, /obj/machinery/iv_drip))
+			fear_amount += 4
```

Comment:
Should probably be more than a regular syringe, and have a bonus of like whenever there's a bag in it.